client/data/disputes/actions.js

```diff
@@ -61,13 +108,265 @@ export function* acceptDispute( id ) {
+				'Please wait until file upload is finished',
+				'woocommerce-payments'
+			)
+		);
+		return;
+	}
+
+	yield updateIsSavingEvidenceForDispute( disputeId, true );
+
+	try {
+		wcpayTracks.recordEvent( 'wcpay_dispute_submit_evidence_clicked' );
+
+		const updatedDispute = yield apiFetch( {
+			path: `${ NAMESPACE }/disputes/${ disputeId }`,
+			method: 'post',
+			data: {
+				// Send full evidence, as submission does not appear to work without new evidence despite being optional.
+				evidence: { ...dispute.evidence, ...evidence },
+				metadata: dispute.metadata,
+				submit: true,
+			},
+		} );
+
+		yield updateEvidenceTransientForDispute( disputeId, {} );
+		yield handleSaveSuccess( disputeId, true );
+		yield updateDispute( updatedDispute );
+	} catch ( err ) {
+		yield handleSaveError( err, true );
+		error = err;
+	} finally {
+		yield updateIsSavingEvidenceForDispute( disputeId, false );
+	}
+
+	return null === error;
+}
+
+export function* saveEvidence( disputeId, evidence ) {
+	let error = null;
+	const dispute = yield select( STORE_NAME ).getDispute( disputeId );
+	const isUploading = yield select(
+		STORE_NAME
+	).getIsUploadingEvidenceForDispute( disputeId );
+
+	// Prevent submit if upload is in progress.
+	if ( some( isUploading ) ) {
+		dispatch( 'core/notices' ).createInfoNotice(
+			__(
+				'Please wait until file upload is finished',
+				'woocommerce-payments'
+			)
+		);
+		return;
+	}
+
+	yield updateIsSavingEvidenceForDispute( disputeId, true );
+
+	try {
+		wcpayTracks.recordEvent( 'wcpay_dispute_save_evidence_clicked' );
+
+		const updatedDispute = yield apiFetch( {
+			path: `${ NAMESPACE }/disputes/${ disputeId }`,
+			method: 'post',
+			data: {
+				// Send full evidence, as submission does not appear to work without new evidence despite being optional.
+				evidence: { ...dispute.evidence, ...evidence },
+				metadata: dispute.metadata,
+				submit: false,
+			},
+		} );
+
+		yield updateEvidenceTransientForDispute( disputeId, {} );
+		yield handleSaveSuccess( disputeId, false );
+		yield updateDispute( updatedDispute );
+	} catch ( err ) {
+		yield handleSaveError( err, false );
+		error = err;
+	} finally {
+		yield updateIsSavingEvidenceForDispute( disputeId, false );
+	}
+
+	return null === error;
+}
+
+const fileSizeExceeded = ( dispute, latestFileSize ) => {
+	const fileSizeLimitInBytes = 4500000;
+	const fileSizes = dispute.fileSize ? Object.values( dispute.fileSize ) : [];
+	const totalFileSize =
+		fileSizes.reduce( ( acc, fileSize ) => acc + fileSize, 0 ) +
+		latestFileSize;
+	if ( fileSizeLimitInBytes < totalFileSize ) {
+		dispatch( 'core/notices' ).createInfoNotice(
+			__(
+				"The files you've attached to this dispute as evidence will exceed the limit for a " +
+					"dispute's total size. Try using smaller files as evidence. Hint: if you've attached " +
+					'images, you might want to try providing them in lower resolutions.',
+				'woocommerce-payments'
+			)
+		);
+		return true;
+	}
+};
```

Comment:
Does this check belong in the File Upload component rather than the upload action? It may not be appropriate to include this in the actions.

Reply:
It makes sense to me in the action, since it seems to pertain more to high-level handling than component workings – what do you think?

Reply:
> it seems to pertain more to high-level handling than component workings

That's a good point 🤔 
I think I agree with you; let's leave this with the actions 👍